internal/db/schema/migrations/postgres/16/04_plugin_host.up.sql

```diff
@@ -129,6 +129,7 @@ begin;
     attributes bytea not null
       constraint attributes_must_not_be_empty
         check(length(attributes) > 0),
+    preferred_endpoints text,
```

Comment:
The benefit of separating this out into a different table is it would allow checks on appropriate prefixes and not nulls.  Having this be a free-form text field which needs to have certain format which is only enforced by go logic has been something we've shied away from in the past.

Reply:
I had it as a different table and had it mostly plumbed through, but when working on the view logic with some help from @jimlambrt we decided it wasn't worth it here. We don't need to check for nulls, validating the data is extremely onerous on the database side (e.g. checking for a valid CIDR or host name, etc.), and it's not a fixed set of data -- it's free text. It's more complicated than what was done in OIDC because it's multiple columns (due to ordering) instead of a single column. It was a huge amount of DB logic, as well as multiple extra calls to the DB to fetch and write and delete values, and it didn't seem worth it just so that the data layer could ensure that there is one of a set of specific prefixes to otherwise free text data.

Reply:
Consider adding a comment on this column to explain why we are breaking from convention a little.

Reply:
Resolving for now while I work on the other approach.